.github/workflows/python-check.yaml

```diff
@@ -114,11 +137,22 @@ jobs:
         uses: actions/cache/save@v4
         with:
           path: icechunk-python/.hypothesis/
-          key: cache-hypothesis-${{ runner.os }}-${{ github.run_id }}
+          key: cache-hypothesis-${{ runner.os }}-${{ github.run_id }}-${{ matrix.deps-version.name }}
 
   xarray-backends:
     runs-on: ubuntu-latest
     needs: [build-wheels]
+    strategy:
+      matrix:
+        xarray-version:
+          - name: minimum
+            version: "2025.01.2"
+            checkout-ref: "v2025.01.2"
+            zarr: "3.0.8"
+          - name: latest
```

Comment:
we do this nightly; I don't think we need it on every PR. Let's just do latest released version.

For the main branch tests, let's use this workflow: https://github.com/earth-mover/icechunk/blob/5a453d6009cadf227d9adffea51b820bc691ee15/.github/workflows/python-upstream.yaml#L203-L207

Reply:
oh i see, the "checkout-ref" is confusing; but it's obnly sued to checkout a version of the backend tests. correct?

Reply:
yes exactly. I'll try renaming it to make it more obvious

Reply:
I was actually able to fully remove it, because now I checkout the tag that matches the installed version.